### PR TITLE
feat(provider): new provider zenvia

### PR DESCRIPTION
**Pull request by Fernando-Erd**
### What change does this PR introduce?

This PR adding a new provider called Zenvia. This provider is very popular in Brazil, Argentina and Mexico

### Why was this change needed?

New provider

### CODE REVIEW

**Comment by Fernando-Erd** [comment]
@michaldziuba03 @jainpawan21 This PR is ready for review.

File: providers/zenvia/src/lib/zenvia.provider.ts [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+  ): Promise<ISendMessageSuccessResponse> {
+    let response = {};
+
+    const contentParse = JSON.parse(options.content);
+
+    switch (contentParse.type_provider) {
+      case 'SMS':
+        response = this.sendSMS(options);
+        break;
+      case 'WHATSAPP':
```
Comment (michaldziuba03) [inline providers/zenvia/src/lib/zenvia.provider.ts:34]: To be honest for me it doesn't feel right - it kinda breaks responsibility of SMS channel.

File: packages/stateless/src/lib/provider/provider.interface.ts [hunk 2]
```diff
@@ -28,7 +28,7 @@ export interface IEmailOptions {
 
 export interface ISmsOptions {
   to: string;
-  content: string;
+  content: any;
```
Comment (michaldziuba03) [inline packages/stateless/src/lib/provider/provider.interface.ts:31]: I don't agree with that change - it should be a string and is expected to be string by other providers.

File: providers/zenvia/src/lib/zenvia.provider.ts [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+  private async sendSMS(
+    options: ISmsOptions
+  ): Promise<ISendMessageSuccessResponse> {
+    const BASE_URL = 'https://api.zenvia.com/v2/channels/sms/messages';
+
+    const contentParse = JSON.parse(options.content);
+
+    const data: ZenviaParams = {
+      from: contentParse.from,
```
Comment (michaldziuba03) [inline providers/zenvia/src/lib/zenvia.provider.ts:137]: I think `from` should be configurable from provider's `options` - it what other providers do.

Example here: https://github.com/novuhq/novu/blob/f2b8089c84ff567d694934253152378a6731c61c/providers/burst-sms/src/lib/burst-sms.provider.ts#L35

File: providers/zenvia/src/lib/zenvia.provider.ts [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+      date: new Date().toISOString(),
+    };
+  }
+
+  private async sendSMS(
+    options: ISmsOptions
+  ): Promise<ISendMessageSuccessResponse> {
+    const BASE_URL = 'https://api.zenvia.com/v2/channels/sms/messages';
+
+    const contentParse = JSON.parse(options.content);
```
Comment (michaldziuba03) [inline providers/zenvia/src/lib/zenvia.provider.ts:134]: `options.content` I believe should be a simple string, not parsable string JSON.

Provider should be configurable by `options`

File: packages/stateless/src/lib/provider/provider.interface.ts (hunk 2 above)
Comment (michaldziuba03) [inline packages/stateless/src/lib/provider/provider.interface.ts:31]: I disagree with that change - content should be always a string and is expected to be a string by other providers.

**Review by michaldziuba03 — COMMENTED** [review]
In my opinion you should change way how you configure behavior of provider - you should use provider's `options` and `options.content` should be a simple string instead parsable stringified JSON.

File: providers/zenvia/src/lib/zenvia.provider.ts (hunk 1 above)
Comment (Fernando-Erd) [reply providers/zenvia/src/lib/zenvia.provider.ts:34]: So should I put WhatsApp sending as a chat category?

**Comment by scopsy** [comment]
Closing for now due to inactivity 🙏

**Pull request closed without merging**